### PR TITLE
Entity todo facilitate deletion

### PR DESCRIPTION
**Pull request by greatbacon**
This PR updates the DeleteEntity flow to properly evict ties related to the entity from the UI stores.

### CODE REVIEW

File: src/lib/vocab/entity/entityMutationHelpers.ts [hunk 1]
```diff
@@ -85,3 +85,38 @@ export const updateTieCaches = (
...
+			const ties = sourceTiesByDestEntityId.get().value.get(dest.source_id);
+			if (ties) {
+				ties.mutate(($ties) => {
+					for (let i = $ties.length - 1; i >= 0; i--) {
+						if ($ties[i].source_id === entity_id) $ties.splice(i, 1);
+					}
+				});
+			}
+		}
+		sourceTiesByDestEntityId.mutate(($v) => $v.delete(entity_id));
```
Comment (ryanatkn) [inline src/lib/vocab/entity/entityMutationHelpers.ts:120]:
```suggestion
		destTiesBySourceEntityId.mutate(($v) => $v.delete(entity_id));
```

File: src/lib/vocab/entity/entityMutations.ts [hunk 2]
```diff
@@ -37,25 +38,24 @@ export const EraseEntities: Mutations['EraseEntities'] = async ({invoke, ui, dis
...
 
-export const DeleteEntities: Mutations['DeleteEntities'] = async ({
-	invoke,
-	params,
-	ui: {entityById, entitiesBySourceId},
-}) => {
+export const DeleteEntities: Mutations['DeleteEntities'] = async ({invoke, params, ui}) => {
 	const result = await invoke();
 	if (!result.ok) return result;
 	//TODO update ties once stores are in place
```
Comment (ryanatkn) [inline src/lib/vocab/entity/entityMutations.ts:44]:
```suggestion
```

File: src/lib/vocab/entity/entityMutationHelpers.ts [hunk 3]
```diff
@@ -85,3 +85,38 @@ export const updateTieCaches = (
 		destTies!.mutate(($v) => $v.push($tie));
 	}
 };
+
+export const evictTieCachesForEntity = (
```
Comment (ryanatkn) [inline src/lib/vocab/entity/entityMutationHelpers.ts:89]: just offering another name if you like it, `evictTiesForEntity` ("update caches" being the thing we use elsewhere that evict may encapsulate)

File: src/lib/vocab/entity/entityMutationHelpers.ts [hunk 4]
```diff
@@ -85,3 +85,38 @@ export const updateTieCaches = (
...
+				});
+			}
+		}
+		sourceTiesByDestEntityId.mutate(($v) => $v.delete(entity_id));
+	}
+
+	const destinations = destTiesBySourceEntityId.get().value.get(entity_id);
+	if (destinations) {
+		for (const dest of destinations.get().value) {
+			const ties = sourceTiesByDestEntityId.get().value.get(dest.source_id);
```
Comment (ryanatkn) [inline src/lib/vocab/entity/entityMutationHelpers.ts:111]: I think this is right? not sure tho

```suggestion
			const ties = sourceTiesByDestEntityId.get().value.get(dest.dest_id);
```

**Review by ryanatkn — APPROVED** [review]
awesome, mind bending but I think this looks good with some possible corrections, also if you want in this PR it'd be easy to call the helper from the `DeleteTie` mutation now as well

File: src/lib/vocab/entity/entityMutationHelpers.ts (hunk 1 above)
Comment (greatbacon) [reply src/lib/vocab/entity/entityMutationHelpers.ts:120]: 🙃

File: src/lib/vocab/entity/entityMutations.ts (hunk 2 above)
Comment (greatbacon) [reply src/lib/vocab/entity/entityMutations.ts:44]: 🧹

File: src/lib/vocab/entity/entityMutationHelpers.ts (hunk 3 above)
Comment (greatbacon) [reply src/lib/vocab/entity/entityMutationHelpers.ts:89]: works fer me!

File: src/lib/vocab/entity/entityMutationHelpers.ts (hunk 4 above)
Comment (greatbacon) [reply src/lib/vocab/entity/entityMutationHelpers.ts:111]: yes, you're totally right!

**Comment by greatbacon** [comment]
@ryanatkn I looked into the deleteTie stuff as well. I think it's technically a different set of logic (or a narrower function I guess) since it's just removing 1 tie from each of the arrays as opposed to having to remove possibly any tie that was pointing at a given entity.

**Pull request merged**